crystalfontz/dbus/map.py

```diff
@@ -0,0 +1,230 @@
+from typing import ClassVar, List, Optional, Protocol, Self, Tuple, Type, Union
+
+from crystalfontz.config import Config
+from crystalfontz.dbus.config import ConfigStruct
+from crystalfontz.device import Device
+from crystalfontz.response import LcdMemory, Pong, UserFlashAreaRead, Versions
+
+
+class TypeProtocol(Protocol):
+    t: ClassVar[str]
+
+
+def t(*args: Union[str, Type[TypeProtocol]]) -> str:
+    type_ = ""
+
+    for arg in args:
+        if isinstance(arg, str):
+            type_ += arg
+        else:
+            type_ += arg.t
+
+    return type_
+
+
+def struct(*args: Union[str, Type[TypeProtocol]]) -> str:
+    return t("(", *args, ")")
+
+
+def array(of: Union[str, Type[TypeProtocol]]) -> str:
+    return f"a{t(of)}"
+
+
+class BytesM:
```

Comment:
I don't actually know how the sdbus library handles `ay`. I'm guessing a `List[int]` will work, but I'm not confident in that.

This is used for the ping command, so it's at least easy to test.